src/libraries/Microsoft.PowerFx.Interpreter/ReflectionFunction.cs

```diff
@@ -235,6 +245,11 @@ private FunctionDescr Scan()
                     {
                         throw new InvalidOperationException($"Cancellation token must be the last argument.");
                     }
+                    else if (parameters[i].ParameterType == typeof(Func<Task<BooleanValue>>))
```

Comment:
>Func<Task<BooleanValue> [](http://example.com/codeflow?start=67&length=23)

If they use other Func<T>, where T != Boolean, would be good to throw a more specific error saying "Only Func<Task<BooleanValue>> is supported for callbacks"
This is step towards once we eventually make Func<Task<T>> work.